proposal/High-availability-for-Container-Registry.md

```diff
@@ -0,0 +1,69 @@
+# High availability for Container Registry
+
+## Summary
+
+Currently the only single point in the cluster is the builtin container registry. Once if the first master node or even only the registry fails with unexpected error, the rest of the nodes in the cluster will fail to pull images. The proposal is to make sure high availability for registry by initializing multiple registries in the cluster, and also distribute offline artifacts to all nodes that with registry role.
+
+## Design Details
+
+1. Added a role named `registry`, and currently only supports apparently specifying a set of host in the `Clusterfile`, the first master node will be selected as the local registry by default if none of the nodes are selected.
+
+a) If no roles are specified, `sealos` will behave as before, aka, this PR does not change all the default behavior of sealos
+b) If we choose some nodes to run as registry, `bootstrap` process will run registry in those hosts before cluster initialization, `run` process will also distribute `registry` dir to those hosts.
+
```

Comment:
Explain that registry does not support scaling and that the registry node must be a node in the cluster?

Reply:
See `Limitations` section.